.github/workflows/CI.yml

```diff
@@ -0,0 +1,113 @@
+          - 3.6
+          - 3.7
+          # Uncomment when ready ;)
+          # - 3.8
+        cfg:
+          - os: ubuntu-latest
+            rdkit: true
+            openeye: false
+
+          - os: ubuntu-latest
+            rdkit: false
+            openeye: true
+
+          - os: ubuntu-latest
+            rdkit: true
+            openeye: true
+
+          - os: macOS-latest
+            rdkit: true
+            openeye: false
+
+    env:
+      CI_OS: ${{ matrix.cfg.os }}
+      RDKIT: ${{ matrix.cfg.rdkit }}
+      OPENEYE: ${{ matrix.cfg.openeye }}
+      PYVER: ${{ matrix.python-version }}
+      OE_LICENSE: ${{ github.workspace }}/oe_license.txt
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Additional info about the build
+      shell: bash
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+    - name: Fix conda permissions in MacOS
+      shell: bash
+      if: startsWith(matrix.cfg.os, 'macOS')
+      run: |
+        sudo chown -R $(id -u):$(id -g) ${CONDA}
+    - name: Configure conda
+      shell: bash
+      run: |
+        . devtools/gh-actions/initialize_conda.sh
+        conda activate
+        conda config --set always_yes yes
+        conda config --add channels omnia --add channels conda-forge --add channels openeye
+        conda install --quiet conda conda-build jinja2 anaconda-client
+        conda update --quiet --all
+        conda info --all
+        conda list
+    - name: Decrypt OpenEye license
+      shell: bash
+      env:
+        ENC_OE_LICENSE: ${{ secrets.OE_LICENSE__EXP_JUN2020 }}
+      run: |
+        echo "${ENC_OE_LICENSE}" > ${OE_LICENSE}
+    - name: Prepare test environment
+      shell: bash
+      run: |
+        . devtools/gh-actions/initialize_conda.sh
+        conda activate
+        python devtools/scripts/create_conda_env.py -n=test -p=$PYVER devtools/conda-envs/test_env.yaml
+        conda activate test
+        if [[ "$RDKIT" == true ]]; then conda install rdkit; fi
+        if [[ "$OPENEYE" == true ]]; then
+          conda install openeye-toolkits
+          python -c "from openeye import oechem; assert oechem.OEChemIsLicensed()"
+        fi
+        python -m pip install --no-deps .
+    - name: Test the package
+      shell: bash
+      run: |
+        . devtools/gh-actions/initialize_conda.sh
+        conda activate test
+        PYTEST_ARGS="-v --ignore=utilities --ignore=examples/deprecated --nbval-lax --cov=openforcefield --cov-config=setup.cfg"
+        if [[ "$RDKIT" == true && "$OPENEYE" == true ]]; then
+          PYTEST_ARGS+=" --ignore=docs --ignore=devtools --doctest-modules"
+        elif [[ "$OPENEYE" == true ]]; then
+          PYTEST_ARGS+=" --ignore=docs --ignore=devtools"
+          PYTEST_ARGS+=" --ignore=examples/check_dataset_parameter_coverage"
+          PYTEST_ARGS+=" --ignore=examples/QCArchive_interface"
+        fi
+        pytest $PYTEST_ARGS
+
+    - name: Codecov
+      uses: codecov/codecov-action@v1
```

Comment:
Does this mean codecov runs on all tests? If we trigger it to only run on a single `"$RDKIT" == true && "$OPENEYE" == true` wouldn't that cover everything?

Reply:
Surprisingly, no. If both toolkits are available, OpenEye will be used preferentially. So there will be lower coverage of RDKit functionality in that build.

Reply:
We can add an `if: <condition>` check to the step if you want! We can also run the coverage on a separate job to make sure it's only run once.

Reply:
Added this in last push to see if it works. Let me know if you agree or if I should revert to no-condition status.

Reply:
@j-wags I'm thinking of something else, not the % of unit tests that it runs on (although that's relevant). I meant if codecov runs on each CI run of the matrix, i.e. all 8 or so checks that CI runs vs. on just one, the reports are probably getting uploaded the same way each time. [Here](https://github.com/mosdef-hub/mbuild/blob/70ee2db5c017aa8dc7737214ea5294fd4ec2d049/azure-pipelines.yml#L78-L84) is what I had in mind (it's AZP but you get the idea, and it is exactly what @jaimergp suggested). I _think_ that codecov will just display whatever report it was sent last, so maybe the last CI check to run and be uploaded?

If all checks were the same, I would push us to pick an arbitrary one and use that as a reference, just to not upload 8x as many reports as we would care about. But, given that the toolkit has slightly different behavior with each other toolkit, we should decide which version we care about. I figure the True/True case is what matters. If we were to care about the code coverage when all RDKit-only or OpenEye-only tests are skipped, that wouldn't be too representative of the coverage of the project as a whole. Here, the advantage of picking just one run in the CI matrix is that we can control which we care about and it should behave the same way each time.

Reply:
Oh, you are right, we are still executing several actions. I might just pop it out to a separate job.

Reply:
> I think that codecov will just display whatever report it was sent last, so maybe the last CI check to run and be uploaded?

Ah, I see. This seems to be the important question. I'm hoping that codecov reports the _union_ of coverage from all tests on a specific commit, instead of having multiple builds for the same commit clobbering each other's reports. 

And indeed, [Codecov's docs](https://docs.codecov.io/docs/merging-reports) say that they do merge reports from multiple builds on the same commit. As further reassurance, it does seem to be grouping [all 8 builds of our recent commits](https://codecov.io/gh/openforcefield/openforcefield/commit/5e6b6dd52980a5e9a0423589c2a6370420e03694/build), and presumably it's merging them for the final readout.

So, if I'm interpreting this correctly, we _do_ want unconditional upload of the coverage report from all builds. But I'm happy to chat tomorrow if I'm misinterpreting the issue.

Reply:
Fixed!

Reply:
I think you're right - may have been different in the past, else I'm not sure where I got that from